react-router/history.d.ts

```diff
@@ -0,0 +1,183 @@
+// Type definitions for history v1.11.1
+// Project: https://github.com/rackt/history
+// Definitions by: Sergey Buturlakin <http://github.com/sergey-buturlakin>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+
+declare namespace HistoryModule {
+
+    // types based on https://github.com/rackt/history/blob/master/docs/Terms.md
+
+    type Action = string
+
+    type BeforeUnloadHook = () => string
+
+    type CreateHistory<T> = (options?: HistoryOptions) => T
+
+    type CreateHistoryEnhancer<T, E> = (createHistory: CreateHistory<T>) => CreateHistory<T & E>
+
+    interface History {
+        listenBefore(hook: TransitionHook): Function
+        listen(listener: LocationListener): Function
+        transitionTo(location: Location): void
+        pushState(state: LocationState, path: Path): void
+        replaceState(state: LocationState, path: Path): void
```

Comment:
When you update this to 0.13.1 or later, you'll want this:

```
        pushState(state: LocationState, path: Path, query?: Query): void
        replaceState(state: LocationState, path: Path, query?: Query): void
        push(path: Path, query?: Query): void
        replace(path: Path, query?: Query): void
```

Note that `push` and `replace` currently don't accept a `query` parameter in the published code, but the published API says they should. I opened an issue for that (https://github.com/rackt/history/issues/144) but left my definitions matching the docs.